src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -6,9 +6,31 @@
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH_DATA = 6;
 
     @Override
     public User register(User user) {
-        return null;
+        if (checkLoginIfPresent(user) != null ) {
+            throw new InvalidDataException("User with this login has been already registered");
+        } else if (user.getAge() == null) {
+            throw new InvalidDataException("Age must not be null");
+        } else if (user.getLogin() == null) {
+            throw new InvalidDataException("Login mustn't be null");
+        } else if (user.getPassword() == null) {
+            throw new InvalidDataException("Password mustn't be null");
+        } else if (user.getLogin().length() < MIN_LENGTH_DATA) {
```

Comment:
The exception message should include the 'MIN_LENGTH_DATA' constant to inform the user about the minimum required length. For example, 'Login must contain at least ' + MIN_LENGTH_DATA + ' characters'.